statefun-flink/statefun-flink-core/src/main/java/org/apache/flink/statefun/flink/core/reqreply/RequestReplyFunction.java

```diff
@@ -228,27 +204,27 @@ private void handleOutgoingDelayedMessages(Context context, InvocationResponse i
   // --------------------------------------------------------------------------------
 
   private void addStates(ToFunction.InvocationBatchRequest.Builder batchBuilder) {
-    for (StateSpec stateSpec : registeredStates) {
-      ToFunction.PersistedValue.Builder valueBuilder =
-          ToFunction.PersistedValue.newBuilder().setStateName(stateSpec.name());
-
-      byte[] stateValue = managedStates.get(stateSpec.name());
-      if (stateValue != null) {
-        valueBuilder.setStateValue(ByteString.copyFrom(stateValue));
-      }
-      batchBuilder.addState(valueBuilder);
-    }
+    managedStates.forEach(
```

Comment:
I think that mangedState should have a `populateBatch` method,
to avoid the forEach allocation.